core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java

```diff
@@ -409,6 +410,17 @@ private static DefaultFunctionResolver rand() {
     );
   }
 
+  /**
+   * Definition of rint(x) function.
+   * Returns the closest whole integer value to x
+   * The supported signature is
+   * BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE -> DOUBLE
```

Comment:
shouldn't it return an integer?  Seems strange to round to integer but turn return a double

Reply:
I had the same thought as you that it would make sense to return an integer but I am following the java definition of the method. My guess is that it returns a double so that it can return numbers big enough to take a double as an input, or that most other functions like this also return doubles. It appears to me that the old engine also returns a double as the Java default so I believe that it should be left as is.